File: core/src/org/rexcellentgames/burningknight/entity/item/accessory/equipable/PoisonBombs.java
package org.rexcellentgames.burningknight.entity.item.accessory.equipable;

import org.rexcellentgames.burningknight.assets.Locale;
import org.rexcellentgames.burningknight.entity.creature.player.Player;

public class PoisonBombs extends Equipable {
	{
		name = Locale.get("poison_bombs");
		description = Locale.get("poison_bombs_desc");
		sprite = "item-poison_bomb";
	}

	@Override
	public void onEquip(boolean load) {
		super.onEquip(load);

		if (this.owner instanceof Player) {
			((Player) this.owner).poisonBombs = true;
		}
	}

	@Override
	public void onUnequip(boolean load) {
		super.onUnequip(load);

		if (this.owner instanceof Player) {
			((Player) this.owner).poisonBombs = false;
		}
	}
}

File: core/src/org/rexcellentgames/burningknight/entity/item/accessory/equippable/PoisonBombs.kt
package org.rexcellentgames.burningknight.entity.item.accessory.equippable

import org.rexcellentgames.burningknight.assets.Locale

class PoisonBombs : Equippable() {
	init {
		name = Locale.get("poison_bombs")
		description = Locale.get("poison_bombs_desc")
		sprite = "item-poison_bomb"
	}

	override fun onEquip(load: Boolean) {
		super.onEquip(load)
		this.owner.poisonBombs = true
	}

	override fun onUnequip(load: Boolean) {
		super.onUnequip(load)
		this.owner.poisonBombs = false
	}

	override fun canBeUpgraded(): Boolean {
		return false
	}

	override fun canBeDegraded(): Boolean {
		return false
	}
}